code/datums/actions/action.dm

```diff
@@ -57,18 +85,26 @@
 	button.desc = desc
 
 	if(action_icon && action_icon_state)
-		button.cut_overlays(TRUE)
-		button.add_overlay(mutable_appearance(action_icon, action_icon_state))
-
+		button.cut_overlay(list(visual_references[VREF_MUTABLE_ACTION_STATE]))
+		var/mutable_appearance/action_appearence = mutable_appearance(action_icon, action_icon_state)
+		action_appearence.layer = HUD_LAYER
+		visual_references[VREF_MUTABLE_ACTION_STATE] = action_appearence
```

Comment:
why regenerate the action appearance every time?cant you cut it, edit icon and icon state and readd?